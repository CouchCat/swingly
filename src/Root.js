import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';

import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
// import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './redux/root-reducer';
import { library } from '@fortawesome/fontawesome-svg-core';
import { 
  faBars,
  faSyncAlt,
  faUserCircle,
  faEdit,
  faSignOutAlt,
  faSignInAlt,
  faHome,
  faCalendar,
  faCalendarDay,
  faCalendarWeek,
  faEllipsisV,
  faCalendarPlus,
  faHatWizard,
  faMoon,
  faSun,
  faChevronLeft
} from '@fortawesome/free-solid-svg-icons';

export default (props) => {
  const { children, initialState = {} } = props;
  const middlewares = [thunk];

  if (process.env.NODE_ENV === 'development') {
    middlewares.push(logger);
  }
  
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middlewares),
  );

  const persistor = persistStore(store);

  library.add(
    faBars,
    faSyncAlt,
    faUserCircle,
    faEdit,
    faSignOutAlt,
    faSignInAlt,
    faHome,
    faCalendar,
    faCalendarDay,
    faCalendarWeek,
    faEllipsisV,
    faCalendarPlus,
    faHatWizard,
    faMoon,
    faSun,
    faChevronLeft
  );

  return (
    <Provider store={store}>
      <BrowserRouter>
        <PersistGate persistor={persistor}>
          {children}
        </PersistGate>
      </BrowserRouter>
    </Provider>
  );
}