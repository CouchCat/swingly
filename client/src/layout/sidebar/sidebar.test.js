import React from 'react';
import { shallow, mount } from 'enzyme';
import ConnectedSidebar, { Sidebar } from './sidebar.component';
import Root from '../../Root';
import { MemoryRouter } from 'react-router-dom';

describe('Sidebar component', () => {
  let wrapper = null;
  let mockProps = null;
  let mockSetOpen = null;
  let mockToggleTheme = null;
  let mockIsDarkMode = null;
  let mockCurrentUser = null;
  let mockOpenSnackbar = null;
  let mockHistory = null;
  
  beforeEach(() => {
    mockSetOpen = jest.fn();
    mockToggleTheme = jest.fn();
    mockHistory = {
      location: {
        pathname: '/'
      }
    }

    mockProps = {
      isOpen: true,
      setOpen: mockSetOpen,
      toggleTheme: mockToggleTheme,
      history: mockHistory
    };

    const initialState = {
      themeMode: { darkMode: false }
    }
    
    wrapper = mount(
      <Root initialState={initialState}>
        <MemoryRouter initialEntries={['/']}>
          <ConnectedSidebar {...mockProps} />
        </MemoryRouter>
      </Root>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render the Sidebar component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the setOpen function on Backdrop click event', () => {
    wrapper.find('Backdrop').simulate('click');
    expect(mockSetOpen).toHaveBeenCalled();
  });

  it('should call the setOpen function on all SideButton (exept last one) click events.', () => {
    const sidebarButtons = wrapper.find('SidebarButton');

    sidebarButtons.forEach(btn => {
      btn.simulate('click');
    })

    expect(mockSetOpen).toHaveBeenCalledTimes(sidebarButtons.length - 1);     
  });

  it('should call the toggleTheme function on last SideButton click event.', () => {
    wrapper = shallow(<Sidebar {...mockProps} />);
    wrapper.find('SidebarButton').last().simulate('click');
    expect(mockToggleTheme).toHaveBeenCalled();
  });

  it('should change text of theme button from "Light Mode" to "Dark Mode".', () => {
    const lastButton = wrapper.find('SidebarButton').last();
    expect(lastButton.text()).toContain('Dark Mode');
  
    lastButton.simulate('click');
    expect(lastButton.text()).toContain('Light Mode');    
  });
  

});