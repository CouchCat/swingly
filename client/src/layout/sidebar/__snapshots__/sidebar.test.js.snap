// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should render the Sidebar component 1`] = `
<Fragment>
  <Backdrop
    data-testid="backdrop"
    isOpen={true}
    onClick={[Function]}
  />
  <SidebarContainer
    data-testid="sidebar-container"
    isOpen={true}
  >
    <SidebarHeader>
      Navigation
    </SidebarHeader>
    <SidebarButton
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      isactive="true"
      onClick={[Function]}
      to="/"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="calendar-week"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      All Events
    </SidebarButton>
    <SidebarButton
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="create-event"
      isactive="false"
      onClick={[Function]}
      to="/event-create"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="calendar-plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      Create Event
    </SidebarButton>
    <SidebarButton
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      isactive="false"
      onClick={[Function]}
      to="/users"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="users"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      Users
    </SidebarButton>
    <SidebarButton
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      isactive="false"
      onClick={[Function]}
      to="/about"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="info-circle"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      About
    </SidebarButton>
    <SidebarButton
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      onClick={[Function]}
      to="/error"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="exclamation-triangle"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      Error Page
    </SidebarButton>
    <SidebarButton
      flat={true}
      onClick={[Function]}
      transparent={true}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="moon"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      Dark Mode
    </SidebarButton>
  </SidebarContainer>
</Fragment>
`;
